### PR TITLE
Updating ec2_ami_find to support product-code and owner-alias filters

### PR DESCRIPTION
**Pull request by peter-murray**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

ec2_ami_find
##### ANSIBLE VERSION

```
devel
```
##### SUMMARY

Added two new filters to the module `product_code` and `owner_alias` to make it possible to find images like CentOS (as per their documentation https://wiki.centos.org/Cloud/AWS) which now uses a product-code.

The addition of the `owner_alias` simplifies the finding of Microsoft based images as a value of `microsoft` can be used to find all images belonging to Microsoft (without having to find their owner id).

Output of the module is completely unchanged by these new values.

To show the usage of the searching for a CentOS image using the `product_code` you can run use `product_code=aw0evgkw8e5c1q413zgy5pjce` and it will return results of:

```
"results": [
        {
            "ami_id": "ami-7abd0209",
            "architecture": "x86_64",
            "block_device_mapping": {
                "/dev/sda1": {
                    "delete_on_termination": false,
                    "encrypted": false,
                    "size": 8,
                    "snapshot_id": "snap-b537027c",
                    "volume_type": "standard"
                }
            },
            "creationDate": "2016-02-26T21:22:50.000Z",
            "description": "CentOS Linux 7 x86_64 HVM EBS 1602",
            "hypervisor": "xen",
            "is_public": true,
            "location": "aws-marketplace/CentOS Linux 7 x86_64 HVM EBS 1602-b7ee8a69-ee97-4a49-9e68-afaee216db2e-ami-d7e1d2bd.3",
            "name": "CentOS Linux 7 x86_64 HVM EBS 1602-b7ee8a69-ee97-4a49-9e68-afaee216db2e-ami-d7e1d2bd.3",
            "owner_id": "679593333241",
            "platform": null,
            "root_device_name": "/dev/sda1",
            "root_device_type": "ebs",
            "state": "available",
            "tags": {},
            "virtualization_type": "hvm"
        }
    ]
```

Similarly using `owner_alias=microsoft` returns results of the form:

```
"results": [
        {
            "ami_id": "ami-3ff3f64b",
            "architecture": "x86_64",
            "block_device_mapping": {
                "/dev/sda1": {
                    "delete_on_termination": false,
                    "encrypted": false,
                    "size": 35,
                    "snapshot_id": "snap-264f064d",
                    "volume_type": "standard"
                },
                "xvdg": {
                    "delete_on_termination": false,
                    "encrypted": false,
                    "size": 60,
                    "snapshot_id": "snap-244f064f",
                    "volume_type": "standard"
                }
            },
            "creationDate": "2012-06-15T18:25:37.000Z",
            "description": "Windows_Server-2008-R2-SP1-English-64Bit-SQL_2008_R2_SP1_Standard-2012.06.12",
            "hypervisor": "xen",
            "is_public": true,
            "location": "microsoft/Windows_Server-2008-R2-SP1-English-64Bit-SQL_2008_R2_SP1_Standard-2012.06.12",
            "name": "Windows_Server-2008-R2-SP1-English-64Bit-SQL_2008_R2_SP1_Standard-2012.06.12",
            "owner_id": "461346954234",
            "platform": "windows",
            "root_device_name": "/dev/sda1",
            "root_device_type": "ebs",
            "state": "available",
            "tags": {},
            "virtualization_type": "hvm"
        },
        {
            "ami_id": "ami-e5d1d491",
            "architecture": "x86_64",
            "block_device_mapping": {
                "/dev/sda1": {
                    "delete_on_termination": true,
                    "encrypted": false,
                    "size": 30,
                    "snapshot_id": "snap-ac9cdec7",
                    "volume_type": "standard"
                }
            },
            "creationDate": "2012-06-12T22:47:30.000Z",
            "description": "",
            "hypervisor": "xen",
            "is_public": true,
            "location": "microsoft/Windows_Server-2008-R2-SP1-English-64Bit-Windows_Media_Services_4.1-2012.06.12",
            "name": "Windows_Server-2008-R2-SP1-English-64Bit-Windows_Media_Services_4.1-2012.06.12",
            "owner_id": "461346954234",
            "platform": "windows",
            "root_device_name": "/dev/sda1",
            "root_device_type": "ebs",
            "state": "available",
            "tags": {},
            "virtualization_type": "hvm"
        },
        {
            "ami_id": "ami-f9d1d48d",
            "architecture": "x86_64",
            "block_device_mapping": {
                "/dev/sda1": {
                    "delete_on_termination": true,
                    "encrypted": false,
                    "size": 35,
                    "snapshot_id": "snap-c29cdea9",
                    "volume_type": "standard"
                }
            },
            "creationDate": "2012-06-12T22:47:02.000Z",
            "description": "",
            "hypervisor": "xen",
            "is_public": true,
            "location": "microsoft/Windows_Server-2008-R2-SP1-English-64Bit-WebMatrix_Development-2012.06.12",
            "name": "Windows_Server-2008-R2-SP1-English-64Bit-WebMatrix_Development-2012.06.12",
            "owner_id": "461346954234",
            "platform": "windows",
            "root_device_name": "/dev/sda1",
            "root_device_type": "ebs",
            "state": "available",
            "tags": {},
            "virtualization_type": "hvm"
        },
        {
            "ami_id": "ami-fbd1d48f",
            "architecture": "x86_64",
            "block_device_mapping": {
                "/dev/sda1": {
                    "delete_on_termination": true,
                    "encrypted": false,
                    "size": 35,
                    "snapshot_id": "snap-aa9cdec1",
                    "volume_type": "standard"
                }
            },
            "creationDate": "2012-06-12T22:47:16.000Z",
            "description": "",
            "hypervisor": "xen",
            "is_public": true,
            "location": "microsoft/Windows_Server-2008-R2-SP1-English-64Bit-WebMatrix_Hosting-2012.06.12",
            "name": "Windows_Server-2008-R2-SP1-English-64Bit-WebMatrix_Hosting-2012.06.12",
            "owner_id": "461346954234",
            "platform": "windows",
            "root_device_name": "/dev/sda1",
            "root_device_type": "ebs",
            "state": "available",
            "tags": {},
            "virtualization_type": "hvm"
        }
    ]
```

### CODE REVIEW

**Comment by peter-murray** [comment]
The CI build is failing as there is no `version_added` on the new options for the module. I have not put them in as I have no idea which version that this change would ship in.

What do you expect me to do with regard to this?

**Comment by gregdek** [comment]
Thanks @peter-murray. To the current maintainers, @tombamford please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/amazon/ec2_ami_find.py [hunk 1]
```diff
@@ -352,12 +366,16 @@ def main():
         filter['architecture'] = architecture
     if hypervisor:
         filter['hypervisor'] = hypervisor
+    if owner_alias:
+        filter['owner_alias'] = owner_alias
```
Comment (manicminer) [inline cloud/amazon/ec2_ami_find.py]: I believe the owner-alias filter is already covered by the `owner` argument as given on original line 366 (new line 384) - which is an amalgamation of account ID and owner alias (you can specify either).

note: @tombamford is my old username

**Comment by peter-murray** [comment]
I have removed the owner_alias from the pull request @manicminer was correct, so this is now just to add support for the `product_code`.

**Comment by gregdek** [comment]
Thanks @peter-murray for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by peter-murray** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @peter-murray. To the current maintainers, @tombamford please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**